platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,210 @@
+                    Collections::unmodifiableMap));
+  }
+
+  Optional<BundleContext> getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(ErrorPageInjector.class);
+    return (cxfBundle != null) ? Optional.of(cxfBundle.getBundleContext()) : Optional.empty();
+  }
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectErrorPage((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  public void init() {
+    checkForMissedServletContexts();
+  }
+
+  private void checkForMissedServletContexts() {
+
+    Optional<BundleContext> optionalBundleContext = getContext();
+    if (!optionalBundleContext.isPresent()) {
+      LOGGER.warn(
+          "Problem retrieving the bundle `Platform :: Error :: Page Injector` after it's just been initialized");
+      return; // This should never happen
+    }
+
+    //  Get all registered servletContexts in OSGI
+    final BundleContext context = optionalBundleContext.get();
+    getServiceReferences(context)
+        .ifPresent(e -> (Arrays.asList(e)).forEach(this::addMissedErrorPage));
+  }
+
+  @SuppressWarnings("unchecked" /* cast of Object[] to ServiceRefrence */)
+  private Optional<ServiceReference<ServletContext>[]> getServiceReferences(BundleContext context) {
+    try {
+      return Optional.ofNullable(
+          (ServiceReference<ServletContext>[])
+              context.getAllServiceReferences(
+                  null, "(" + Constants.OBJECTCLASS + "=javax.servlet.ServletContext)"));
+
+    } catch (InvalidSyntaxException | ClassCastException e) {
+      LOGGER.error(
+          "Problem getting ServletContexts from OSGI, One of the servlets running might print stack traces to the browser. A system restart is recommended. See debug logs for additional details.");
+      LOGGER.debug("Problems getting ServletContexts from OSGI Details:", e);
+      return Optional.empty();
+    }
+  }
+
+  private void addMissedErrorPage(ServiceReference<ServletContext> reference) {
+
+    final Bundle refBundle = reference.getBundle();
+    final BundleContext bundlectx = refBundle.getBundleContext();
+    ServletContext service = bundlectx.getService(reference);
+
+    //  Need to use the service to get the errorPageErrorHandler through the
+    // ServletContextHandler
+    Optional<ServletContextHandler> optionalHttpContext = getHTTPContext(service, refBundle);
+
+    if (!optionalHttpContext.isPresent()) return;
+
+    ServletContextHandler httpContext = optionalHttpContext.get();
+
+    ErrorPageErrorHandler errorPageErrorHandler = null;
```

Comment:
✏️ This can be simplified a bit by moving this declaration inline and moving the conditional block on lines 135-138 inside the `try` block.

Reply:
I like the suggestion but I'm not a huge on the idea of putting more inside of try blocks than necessary 👍